### ISSUE TITLE
`beeline.flush()` needs a timeout parameter

### ISSUE THREAD

**Issue by djsweet** [honeycombio/beeline-nodejs#201]
On May 13 2020, Honeycomb suffered an outage. Because of the outage, our instances of Google Cloud Functions effectively stalled in `beeline.flush()`. This not only consumed CPU time in Cloud Functions, it also caused us to go over our execution time quotas, resulting in a full outage on our part.

The core of the issue is that we were getting stuck in `beeline.flush()`. If we could just abandon the `flush()` method in a reliable manner, we would not have suffered an outage at the same time as Honeycomb.

We're going to try something with `Promise.race()` as a short-term workaround but we're concerned that we're going to leak network connections this way.

**Comment by toshok** [comment]
Once that PR lands and we bump the libhoney dep here, the `timeout: ` config parameter should be available in the beeline, so it'll be the equivalent of:

```
require("honeycomb-beeline")({
  timeout: 15000,
   ... rest of options...
});
```

**Comment by djsweet** [comment]
Just having a guaranteed 60s timeout would be a huge improvement on its own. What we saw is that we would block in `beeline.flush()` until GCP terminated our functions, which may have taken up to 540 seconds depending on our Cloud Function configuration. Is there some way to upgrade the `libhoney` dependency out of band so that we can get that 60 seconds timeout "ahead of time"?

**Comment by vreynolds** [comment]
Hello,

We will be closing this issue as it is a low priority for us. It is unlikely that we'll ever get to it, and so we'd like to set expectations accordingly.

As we enter 2022 Q1, we are trimming our OSS backlog. This is so that we can focus better on areas that are more aligned with the OpenTelemetry-focused direction of telemetry ingest for Honeycomb.

If this issue is important to you, please feel free to ping here and we can discuss/re-open.
